### PR TITLE
Finished task Lvjavj 27v3 (OSBB address should use Address Classificator (Region/City/Street))

### PR DESCRIPTION
**Pull request by YuriPushchalo**
**BackEnd changes:**
Osbb entity - "street" attribute added (appropriate liquibase Changheset changed)
db.changelog-master.json - order of table creation changed
import.sql - Regions names chahged
AddressController - handling of bad request results added

**FrontEnd changes:**
i18n files changed
osbb.component.html, osbb.component.ts - changes provide address colunm correct display new address functional
osbb-modal.html, osbb-modal.component.ts - modal window for OSBB adding/editing changed, providing Address Classificator functional using

some minor changes (refactoring)

### CODE REVIEW

File: persistence/src/main/resources/db/changelog/osbb/create-osbb-table.json [hunk 1]
```diff
@@ -159,6 +165,28 @@
...
+              "onUpdate": "NO ACTION",
+              "referencedColumnNames": "id",
+              "referencedTableName": "streets"
+            }
+          }
+        ]
+      }
     }
   ]
 }
```
Comment (thrytsko-amplify) [inline persistence/src/main/resources/db/changelog/osbb/create-osbb-table.json]: add empty line

File: web/src/main/resources/public/src/shared/services/address.service.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     }           
+
+     getStreetById(streetID: number): Observable<Street> {
+        console.log('Get street: ');
+        let url = this.addressUrl + '/street/id/' + streetID;
+        return this.http.get(url)
+            .map((res:Response)=> res.json())
+            .catch((error)=>Observable.throw(error)); 
+     }           
+}
```
Comment (thrytsko-amplify) [inline web/src/main/resources/public/src/shared/services/address.service.ts]: add empty line

File: web/src/main/resources/public/src/shared/models/addressDTO.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+     id: number;
+     name: string;
+     region: Region;
+}
+
+export class Street {
+     id: number;
+     name: string;
+     city: City;
+}
```
Comment (thrytsko-amplify) [inline web/src/main/resources/public/src/shared/models/addressDTO.ts]: add empty line

File: web/src/main/resources/public/src/app/admin/components/osbb/osbb_form/osbb-modal1.html [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-default" (click)="clearForm();modalWindow.hide()">{{'cancel' | translate | capitalize}}</button>
+                <button type="button" (click)="saveButtonAction(inputLogo)" class="btn btn-default">
+                    {{'save' | translate | capitalize}}
+                </button>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (thrytsko-amplify) [inline web/src/main/resources/public/src/app/admin/components/osbb/osbb_form/osbb-modal1.html]: add empty line

File: web/src/main/resources/public/src/app/admin/components/osbb/osbb.service.ts [hunk 5]
```diff
@@ -109,4 +111,5 @@ export class OsbbService {
         console.log('HandleError', error);
         return Promise.reject(error.message || error);
     }
+         
 }
```
Comment (thrytsko-amplify) [inline web/src/main/resources/public/src/app/admin/components/osbb/osbb.service.ts]: add empty line

**Comment by YuriPushchalo** [comment]
All fixed according to **thrytsko-amplify** remarks.

File: web/src/main/java/com/softserve/osbb/controller/AddressController.java [hunk 6]
```diff
@@ -40,19 +41,33 @@
     @RequestMapping(value = "/city/{id}", method = RequestMethod.GET)
     public ResponseEntity<List<City>> getAllCitiesOfRegion(@PathVariable("id") Integer id) {
         logger.info("Get all cities of region: ");
-        return new ResponseEntity<>(addressService.getAllCitiesOfRegion(id), HttpStatus.OK);
+        if (addressService.getRegionById(id) == null) {
+            return new ResponseEntity<>(new ArrayList<City>(), HttpStatus.NOT_FOUND);
+        } else {
+            return new ResponseEntity<>(addressService.getAllCitiesOfRegion(id), HttpStatus.OK);
```
Comment (KostyantynPanchenko) [inline web/src/main/java/com/softserve/osbb/controller/AddressController.java]: Two return statements - not good.
Please change your code like this:

HttpStatus status = HttpStatus.OK;
List&lt;City&gt; cities = new ArrayList<>();
if (addressService.getRegionById(id) == null) {
    status = HttpStatus.NOT_FOUND;
} else {
    cities.addAll(addressService.getAllCitiesOfRegion(id));
}
return new ResponseEntity<>(cities, status);

File: web/src/main/resources/public/src/app/admin/components/osbb/osbb.component.ts [hunk 7]
```diff
@@ -45,6 +45,15 @@ export class OsbbComponent implements OnInit {
         this.updatedOsbb = osbb;
     }
 
+    composeAddress(element: IOsbb): string {
+        if (element.street != null) {
+        return (element.street.city.region.name + ', м.' + element.street.city.name +
```
Comment (KostyantynPanchenko) [inline web/src/main/resources/public/src/app/admin/components/osbb/osbb.component.ts:50]: Rewrite to single return statement

**Review by KostyantynPanchenko — APPROVED** [review]
+

**Review by sergvasylchak — APPROVED** [review]

File: web/src/main/resources/public/src/app/admin/components/osbb/osbb_form/osbb-modal.component.ts [hunk 8]
```diff
@@ -5,14 +5,19 @@ import { TranslatePipe } from "ng2-translate";
 import { CapitalizeFirstLetterPipe } from "../../../../../shared/pipes/capitalize-first-letter";
 import { IOsbb, Osbb } from "../../../../../shared/models/osbb";
 import { Attachment } from "../../../../../shared/models/attachment";
+import { SelectItem } from "../../../../../shared/models/ng2-select-item.interface";
```
Comment (Kvolotc) [inline web/src/main/resources/public/src/app/admin/components/osbb/osbb_form/osbb-modal.component.ts]: delete this import

File: web/src/main/resources/public/src/app/admin/components/osbb/osbb.service.ts [hunk 9]
```diff
@@ -1,20 +1,22 @@
...
+import {Observable} from 'rxjs/Observable';
 
 import "rxjs/add/operator/map";
 import "rxjs/add/operator/toPromise";
+import 'rxjs/add/operator/catch';
 
 import { IOsbb } from "../../../../shared/models/osbb";
 import { OsbbDTO } from "../../../../shared/models/osbbDTO";
 import ApiService = require("../../../../shared/services/api.service");
+import { SelectItem } from "../../../../shared/models/ng2-select-item.interface";
```
Comment (Kvolotc) [inline web/src/main/resources/public/src/app/admin/components/osbb/osbb.service.ts]: delete this import

**Review by OlegKykena — APPROVED** [review]

**Pull request merged**